include/clang/AST/Type.h

```diff
@@ -3689,6 +3693,33 @@ class UnresolvedUsingType : public Type {
 };
 
 
+class TypeVariableType : public Type, public llvm::FoldingSetNode {
+  unsigned int deBruijnDepth;
```

Comment:
Please look at how parameter variables are being symbolically represented elsewhere in clang.  They have scope depth and positional offset information.  We probably want to treat type variables in a similar fashion.